ui/core-react/src/core-react/l10n/usePackageTranslation.tsx

```diff
@@ -0,0 +1,73 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import * as React from "react";
+import { useLocalization } from "./LocalizationProvider";
+
+/** Used by AppUI packages internally to define package specific `useTranslation` hook.
+ * Uses a localization provider to translate the provided key with a specified namespace. Multiple fallback layers are used to resolve the translation:
+ * - Return value of a specified `fallback` function.
+ * - Default value looked up from a specified `defaults` object.
+ * - Lastly `key` argument of the translation function is returned.
+ * @internal
+ */
+export function usePackageTranslation({
+  namespace,
+  fallback,
+  defaults,
+}: {
+  namespace: string;
+  fallback: (key: string) => string | undefined;
+  defaults: object;
+}) {
+  const localization = useLocalization();
+  const [registered, setRegistered] = React.useState(false);
+  React.useEffect(() => {
+    if (!localization) return;
+
+    let ignore = false;
+    setRegistered(false);
+    void (async () => {
+      await localization.registerNamespace(namespace);
```

Comment:
does this just become a no-op if its already registered? seems like this would get called quite a bit

Reply:
yes, `registerNamespace` returns the same promise for each namespace.